pkg/vault/helpers.go

```diff
@@ -94,3 +98,46 @@ func ReadBasicAuth(client Client, path string) (*config.BasicAuth, error) {
 		Password: passwordStr,
 	}, nil
 }
+
+// ToURI constructs a vault: URI for the given path and key
+func ToURI(path string, key string) string {
+	return fmt.Sprintf("vault:%s:%s", path, key)
+}
+
+// ReplaceURIs will replace any vault: URIs in a string, using the vault client
+func ReplaceURIs(s string, client Client) (string, error) {
+	var err error
+	answer := vaultURIRegex.ReplaceAllStringFunc(s, func(found string) string {
+		// Stop once we have an error
+		if err == nil {
+			pathAndKey := strings.TrimSuffix(strings.TrimPrefix(strings.TrimPrefix(found, "vault:"), "\""), "\"")
+			parts := strings.Split(pathAndKey, ":")
+			if len(parts) != 2 {
+				err = errors.Errorf("cannot parse %s as path:key", pathAndKey)
+				return ""
+			}
+			secret, err1 := client.Read(parts[0])
+			if err1 != nil {
+				err = err1
+				return ""
+			} else {
+				if v, ok := secret[parts[1]]; !ok {
+					err = errors.Errorf("unable to find %s in secret at %s", parts[1], parts[0])
+					return ""
+				} else {
```

Comment:
Also you don't need this `else`.

Reply:
Hound thought that too, but I do as I need the value of `v` in the else. Bug in the linter ;-)